src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -554,8 +554,7 @@ public function dumpFile($filename, $content)
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
-        // Ignore for filesystems that do not support umask
-        @chmod($tmpFile, 0666);
+        @chmod($tmpFile, 0666 & ~umask());
```

Comment:
Can you also adjust the test case to test this behavior? (be aware of windows not having umask)